src/configurationProvider.ts

```diff
@@ -213,8 +212,90 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
         return Object.keys(config).filter((key: string) => key !== "noDebug").length === 0;
     }
 
-    private async chooseMainClass(folder: vscode.WorkspaceFolder | undefined): Promise<IMainClassOption | undefined> {
-        const res = await resolveMainClass(folder ? folder.uri : undefined);
+    private isEmptyArray(configItems: any): boolean {
+        return !Array.isArray(configItems) || !configItems.length;
+    }
+
+    private async resolveLaunchConfig(folder: vscode.Uri | undefined, config: vscode.DebugConfiguration): Promise<IMainClassOption> {
+        if (!config.mainClass) {
+            return await this.promptMainClass(folder);
+        }
+
+        const containsExternalClasspaths = !this.isEmptyArray(config.classPaths) || !this.isEmptyArray(config.modulePaths);
+        const validationResponse = await validateLaunchConfig(folder, config.mainClass, config.projectName, containsExternalClasspaths);
+        if (!validationResponse.mainClass.isValid || !validationResponse.projectName.isValid) {
+            return await this.fixMainClass(folder, config, validationResponse);
+        }
+
+        return {
+            mainClass: config.mainClass,
+            projectName: config.projectName,
+        };
+    }
+
+    private async fixMainClass(folder: vscode.Uri | undefined, config: vscode.DebugConfiguration, validationResponse: ILaunchValidationResponse):
+        Promise<IMainClassOption | undefined> {
```

Comment:
A promise should be resolved with real results instead of undefined. In the case that we are not able to resolve some real results, it looks like an exception case.

Reply:
when user cancels the fix dropdown, should return `undefined` instead of an exception. because it's user consent behavior.